### PR TITLE
feat: add reports filters

### PR DESCRIPTION
**Pull request by fcoronelmakingsense**
- Add new page Reports
- Add new componet reports filters for the domains and pages filters
- Add texts
- Add fake domains and fake pages
- TO DO add test

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ class App extends Component {
         {isLoggedIn ? (
           <>
             <Header />
-            <img src={logo} alt="logo" />
+            <Reports />
```
Comment (mvillaseco) [inline src/App.js]: Loved this 👍

File: src/App.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
```
Comment (andresmoschini) [inline src/App.js]: Did you delete the file?

File: src/components/Reports/ReportsFilters/ReportsFilters.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          <FormattedMessage id="reports_filters.description" />
+        </p>
+        <form action="#" className="form-filters">
+          <fieldset className="filter">
+            <label htmlFor="domain">
+              <FormattedMessage id="reports_filters.domain" />
+            </label>
+            <span className="dropdown-arrow" />
+            <select
+              onChange={changeDomain}
```
Comment (andresmoschini) [inline src/components/Reports/ReportsFilters/ReportsFilters.js]: What do you think about extracting the id here? in that way, outside components do not need to know about elements, events, etc, for example:

```suggestion
              onChange={(event) => changeDomain(parseInt(event.target.value)}
```

File: src/components/Reports/ReportsFilters/ReportsFilters.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+              id="domain"
+              className=""
+              name=""
+              disabled={domains.length === 0}
```
Comment (andresmoschini) [inline src/components/Reports/ReportsFilters/ReportsFilters.js]: Since outside you are writing `if (domain.length)`, what do you think about this:
```suggestion
              disabled={!domains.length}
```

File: src/components/Reports/ReportsFilters/ReportsFilters.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+          <fieldset className="filter">
+            <label htmlFor="domain">
+              <FormattedMessage id="reports_filters.domain" />
+            </label>
+            <span className="dropdown-arrow" />
+            <select
+              onChange={changeDomain}
+              id="domain"
+              className=""
+              name=""
```
Comment (andresmoschini) [inline src/components/Reports/ReportsFilters/ReportsFilters.js]: Is adding these empty values necessary?

File: src/components/Reports/ReportsFilters/ReportsFilters.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+              </FormattedMessage>
+              <FormattedMessage id="reports_filters.rank_time_item2">
+                {(message) => <option value="2">{message}</option>}
+              </FormattedMessage>
+              <FormattedMessage id="reports_filters.rank_time_item3">
+                {(message) => <option value="3">{message}</option>}
+              </FormattedMessage>
+              <FormattedMessage id="reports_filters.rank_time_item4">
+                {(message) => <option value="4">{message}</option>}
+              </FormattedMessage>
```
Comment (andresmoschini) [inline src/components/Reports/ReportsFilters/ReportsFilters.js]: What do you think about hardcoding these entries in a const before `return` statement and using a _map_ here?

File: src/components/Reports/ReportsFilters/ReportsFilters.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+              name=""
+              disabled={domains.length === 0}
+            >
+              {domains.map((domain, index) => (
+                <option key={index} value={domain.id}>
+                  {domain.name}
+                </option>
+              ))}
+            </select>
+            {domains.length ? (
```
Comment (andresmoschini) [inline src/components/Reports/ReportsFilters/ReportsFilters.js]: Since this section is related to `domainSelected`, what do you think about using `domainSelected` as a flag in place of `domains.length`?

```suggestion
            {domainSelected ? (
```

File: src/components/Reports/Reports.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        domains: domains,
+        domainSelected: domainSelected,
+        pages: pages,
+        pageSelected: pageSelected,
+      });
+    }
+  }
+
+  changeDomain = async (event) => {
+    const id = parseInt(event.target.value);
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]: Please, see [this comment](https://github.com/FromDoppler/doppler-webapp/pull/55/files#r264904246) in ReportsFilter.

File: src/components/Reports/Reports.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    const domainFound = this.state.domains.find(function(item) {
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]:
```suggestion
    const domainFound = this.state.domains.find((item) => item.id === id);
```

File: src/components/Reports/Reports.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+  render() {
+    return (
+      <>
+        <ReportsFilters
+          changeDomain={this.changeDomain}
+          domains={this.state.domains}
+          domainSelected={this.state.domainSelected}
+          pages={this.state.pages}
+        />
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]: Since you are updating the state with _pageSelected_, I think that you should pass it here and make honor to it in ReportsFilter.

File: src/components/Reports/ReportsFilters/ReportsFilters.js [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+                </span>
+              </span>
+            ) : null}
+          </fieldset>
+          <fieldset className="filter">
+            <label htmlFor="pages">
+              <FormattedMessage id="reports_filters.pages" />
+            </label>
+            <span className="dropdown-arrow" />
+            <select id="pages" className="" name="pages" disabled={pages.length === 0}>
```
Comment (andresmoschini) [inline src/components/Reports/ReportsFilters/ReportsFilters.js]:
```suggestion
            <select id="pages" className="" name="pages" disabled={!pages.length}>
```

File: src/components/Reports/Reports.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import ReportsFilters from './ReportsFilters/ReportsFilters';
+import { getDomains, getPagesByDomainId } from './ReportsService';
```
Comment (andresmoschini) [inline src/components/Reports/Reports.js]: ❤️

**Comment by andresmoschini** [comment]
Really nice work @fcoronelmakingsense! It is very tidy and modular. It would be really nice if you add some tests for it before merging.

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 6 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: maybe is better has and static information if can

File: src/App.js [hunk 13]
```diff
@@ -8,6 +7,7 @@ import Footer from './components/Footer/Footer';
 import axios from 'axios';
 import { HttpDopplerLegacyClient } from './services/doppler-legacy-client';
 import { OnlineSessionManager } from './services/session-manager';
+import Reports from './components/Reports/Reports';
```
Comment (cbernat) [inline src/App.js]: Why there is a reports inside reports here? What's the logic here?

Comment (fcoronelmakingsense) [reply src/App.js]: Reports it's the folder contain all reports components. And the other reports it's the Reports page

File: src/App.js (hunk 1 above)
Comment (cbernat) [reply src/App.js]: Me too :)

**Comment by cbernat** [comment]
👍

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 6 above)
Comment (andresmoschini) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: As I said in our call, I think that it could be clean, but I have not a strong opinion on it.

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 5 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: fixed

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 7 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: fixed

File: src/components/Reports/Reports.js (hunk 8 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/Reports.js]: fixed

File: src/components/Reports/Reports.js (hunk 9 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/Reports.js]: fixed

File: src/components/Reports/Reports.js (hunk 10 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/Reports.js]: fixed

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 11 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: fixed

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 4 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: fixed

File: src/components/Reports/ReportsFilters/ReportsFilters.js (hunk 3 above)
Comment (fcoronelmakingsense) [reply src/components/Reports/ReportsFilters/ReportsFilters.js]: fixed

File: src/App.js (hunk 2 above)
Comment (fcoronelmakingsense) [reply src/App.js]: now yes

**Comment by fcoronelmakingsense** [comment]
Add test

File: src/components/Reports/Reports.test.js [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+describe('Reports page', () => {
+  afterEach(cleanup);
+
+  beforeEach(() => {
+    getDomains.mockClear();
+    getPagesByDomainId.mockClear();
+  });
+
+  it('load domains without pages', () => {
```
Comment (andresmoschini) [inline src/components/Reports/Reports.test.js]: Is this test redundant? 

Could you rephrase it as an assertion sentence, like _"it (Reports page) should render domains without pages"_

Comment (fcoronelmakingsense) [reply src/components/Reports/Reports.test.js]: fixed
